### ISSUE TITLE
Unauthorized error when running as service or cronjob

### ISSUE THREAD

**Issue by ricwtk** [abraunegg/onedrive#858]
### Bug Report Details ###
**Describe the bug**
onedrive gives unauthorized error when running as a service or through cronjob. onedrive synchronizes fine if manually triggered using `onedrive --synchronize`.

**Application and Operating System Details:**
*   OS: Linux zenux 5.5.13-1-MANJARO #1 SMP PREEMPT Wed Mar 25 17:14:28 UTC 2020 x86_64 GNU/Linux, Manjaro
*   Are you using a headless system (no gui) or with a gui installed? GUI, Gnome Desktop
*   OneDrive Account Type: OneDrive for Business
*   DMD or LDC compiler version: v2.091.0
*   Application configuration:
``` 
onedrive version                    = v2.3.3-14-gbac9ab4
Config path                         = /home/ricwtk/.config/onedrive
Config file found in config path    = true
Config option 'check_nosync'        = false
Config option 'sync_dir'            = /mnt/data/Linux/OneDrive
Config option 'skip_dir'            = 
Config option 'skip_file'           = ~*|.~*|*.tmp
Config option 'skip_dotfiles'       = false
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 45
Config option 'min_notify_changes'  = 5
Config option 'log_dir'             = ~/.log/onedrive_official
Config option 'sync_root_files'     = false
Selective sync configured           = false
```
*   Curl Version:
```
curl 7.69.1 (x86_64-pc-linux-gnu) libcurl/7.69.1 OpenSSL/1.1.1e zlib/1.2.11 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh2/1.9.0 nghttp2/1.40.0
Release-Date: 2020-03-11
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets
```

**Note:** Please generate a full debug log as per [https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support](https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support) and email to support@mynas.com.au

**To Reproduce**
Steps to reproduce the behavior if not causing an application crash:

*systemd service*
1. enable onedrive as service: `systemctl --user enable onedrive`.
2. monitor the service using `journalctl --user-unit onedrive -f`.
3. The log shows a HTTP 401 Unauthorized error has been received.

*cronjob*
1. use `crontab -e` to automatically trigger synchronization and save the log: `onedrive --sycnhronize --verbose >> ~/onedrive.log`.
2. The log shows a HTTP 401 Unauthorized error has been received.

*manual trigger*
1. manually trigger the synchronization process from terminal: `onedrive --synchronize --verbose`.
2. The synchronization is successful.

**Complete Verbose Log Output**
Application Log Output:
```bash
Using Config Dir: /home/ricwtk/.config/onedrive
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /mnt/data/Linux/OneDrive

ERROR: OneDrive returned an error with the following message:
  Error Message: HTTP request returned status code 400 (Bad Request)
  Error Reason:  AADSTS70000: Provided grant is invalid or malformed.
Trace ID: a3ab6fa5-7e63-4c3e-8af2-b79fb6590200
Correlation ID: 8bcf9f99-bcb1-4fe3-94be-f96f1db36d87
Timestamp: 2020-04-04 06:40:02Z

ERROR: OneDrive returned a 'HTTP 401 Unauthorized' - Cannot Initialize Sync Engine
ERROR: Check your configuration as your refresh_token may be empty or invalid. You may need to issue a --logout and re-authorise this client.
```

### Bug Report Checklist ###
*   [X] Detailed description
*   [X] Reproduction steps (if applicable)
*   [X] Verbose Log Output

**Comment by abraunegg** [comment]
@ricwtk 
Please validate what version you have installed in all paths, as your `--display-config` output is detailing `v2.3.3-14-gbac9ab4` however your verbose log output is detailing what would be present when running `v2.4.x`

Please can you confirm all application version you are running. You might need to search for all binaries called 'onedrive' to determine where these have all been installed, and fix this up.

Secondly, regarding the error message in your verbose log, please refer to https://github.com/abraunegg/onedrive/issues/835

This was a necessary change for the client. As per the error message that is displayed:
```
You may need to issue a --logout and re-authorise this client.
```

**Comment by ricwtk** [comment]
Thanks @abraunegg. I found out that I have two onedrive, one in `/usr/local/bin/` and the other in `/usr/bin/`. 

If I'm not wrong, `/usr/local/bin/onedrive` was installed when I first switched to Manjaro, and installed onedrive manually from aur repo. Later when I started using `yay`, it installed onedrive into `/usr/bin/onedrive`, which I assumed the previous installation was replaced. That's why even though previously I when I issued `--logout` as user, it only re-authorised for the application ID of the older version from `/usr/local/bin/onedrive`.

I removed `/usr/local/bin/onedrive` and it seems to be working fine now. Thank you and keep up the good work.
